From: Josef Bacik <jbacik@fb.com>
Date: Fri, 25 Mar 2016 13:25:59 -0400
Patch-mainline: 4.8
Git-commit: d38b349c39a9feb1aca82c3280b9b1234497d582
References: bsc#1012452
Subject: [PATCH] Btrfs: don't bother kicking async if there's nothing
 to reclaim

We do this check when we start the async reclaimer thread, might as well check
before we kick it off to save us some cycles.  Thanks,

Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: David Sterba <dsterba@suse.com>
---
 fs/btrfs/extent-tree.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/fs/btrfs/extent-tree.c
+++ b/fs/btrfs/extent-tree.c
@@ -4931,6 +4931,9 @@ static inline int need_do_async_reclaim(
 	if ((space_info->bytes_used + space_info->bytes_reserved) >= thresh)
 		return 0;
 
+	if (!btrfs_calc_reclaim_metadata_size(fs_info->fs_root, space_info))
+		return 0;
+
 	return (used >= thresh && !btrfs_fs_closing(fs_info) &&
 		!test_bit(BTRFS_FS_STATE_REMOUNTING, &fs_info->fs_state));
 }
